Mathlib/Analysis/Convex/EGauge.lean

```diff
@@ -69,6 +74,21 @@ lemma le_egauge_inter (s t : Set E) (x : E) :
     egauge 𝕜 s x ⊔ egauge 𝕜 t x ≤ egauge 𝕜 (s ∩ t) x :=
   max_le (egauge_anti _ inter_subset_left _) (egauge_anti _ inter_subset_right _)
 
+lemma le_egauge_pi {ι : Type*} {E : ι → Type*} [∀ i, SMul 𝕜 (E i)] {I : Set ι} {i : ι}
+    (hi : i ∈ I) (s : ∀ i, Set (E i)) (x : ∀ i, E i) :
+    egauge 𝕜 (s i) (x i) ≤ egauge 𝕜 (I.pi s) x :=
+  MapsTo.egauge_le _ (Pi.evalMulActionHom i) (fun x hx ↦ by exact hx i hi) _
+
+variable {F : Type*} [SMul 𝕜 F]
+
+lemma le_egauge_prod_left (s : Set E) (t : Set F) (a : E) (b : F) :
+    egauge 𝕜 s a ≤ egauge 𝕜 (s ×ˢ t) (a, b) :=
+  MapsTo.egauge_le 𝕜 (MulActionHom.fst 𝕜 E F) mapsTo_fst_prod (a, b)
+
+lemma le_egauge_prod_right (s : Set E) (t : Set F) (a : E) (b : F) :
+    egauge 𝕜 t b ≤ egauge 𝕜 (s ×ˢ t) (a, b) :=
+  MapsTo.egauge_le 𝕜 (MulActionHom.snd 𝕜 E F) mapsTo_snd_prod (a, b)
```

Comment:
I'd be inclined to merge these into a single
```
lemma le_egauge_prod (s : Set E) (t : Set F) (a : E) (b : F) :
    max (egauge 𝕜 s a) (egauge 𝕜 t b) ≤ egauge 𝕜 (s ×ˢ t) (a, b) :=
```
which is all you end up needing below anyway